### ISSUE TITLE
Can not search metadata for files having common attributes

### ISSUE THREAD

**Issue by hanytran** [irods-contrib/metalnx-web#322]
In metadata search, when there are files having common attributes and owned by more than one user. Once one of them tries to search, no results return because it failed at getting permission in MetadataService of metalnx.
Error message: "Could not get permissions for current user: unable to find file under path"
iCommand in docker container is working just fine.

**Comment by korydraughn** [comment]
What version of iRODS, OS, and Metalnx are you running?

Please provide steps showing how to reproduce this?

**Comment by hanytran** [comment]
iRODS 4.2.6, OS: ubuntu 20.04, and Metalnx 2.4.0

- user1 uploads file1 having metadata (att1,  val1)
- user1 searches metadata with att1 -> found 1
- user2 uploads file2 having metadata (att1,  val2)
- user1 searches metadata with att1 -> nothing returns
- user2 searches metadata with att1 -> nothing returns

**Comment by korydraughn** [comment]
Can you try this on a later version of iRODS and Metalnx?

The issue you're seeing may have been resolved in a later release.

Also, we haven't released Ubuntu 20 packages for that version of iRODS. I'm curious as to how you made that work.

**Comment by hanytran** [comment]
I built by myself the docker image for iRODS 4.2.6 and used it in a Docker compose for this whole setup (iRODS, metalnx, jupyterhub, keycloak).

I tried yesterday to install iRODS 4.3 from your link you gave (https://github.com/irods/irods_development_environment) but it was not successfully yet. I found it's a bit complex because we need to build core and runner with other settings...

**Comment by korydraughn** [comment]
Ok. I'll investigate this issue and get back to you once I know more.

As for the development environment, that sounds right. We don't expose compilation options. Will have to think about that one.

**Comment by hanytran** [comment]
Thanks.
I really want to use iRODS 4.3 in the docker compose but it's not that easy to install and set up. I think in this version it's quite cool when one can write rules in python 3 and can also work with metalnx 2.6. Besides it seems many bugs have been already fixed.

**Comment by korydraughn** [comment]
You may want to consider joining the iRODS google group at https://groups.google.com/g/iROD-Chat

There, you'll have access to the community and they can provide additional support and guidance to various use-cases.

We'll look into what we can do regarding a simple docker compose project.

**Comment by korydraughn** [comment]
I could not reproduce what you're seeing in later versions of iRODS and Metalnx.

I was able to search and find data objects using shared attribute names without any issues. This issue was likely resolved by the consolidation of the search interfaces (Property search and Metadata/AVU search).

Closing this as the solution to your problem is to upgrade.

---

I've created an issue for the development environment regarding your comments about compilation. See https://github.com/irods/irods_development_environment/issues/71
